[inductor] Fix no-xdim reductions

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103527

Fixes #103481

Normally triton tensors have shape `[XBLOCK, RBLOCK]`, or some variation where
the lengths are 1 but the number of dimensions is the same. The `no_x_dim`
change in addition to removing the x dimension, also removed the r dimension
from certain values such as the results of reductions and the `xindex` variable.

This fixes those two cases to correctly produce tensors of shape `[1]`,
equivalent to the old shape `[XBLOCK, 1]` with the x-dimension dropped.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @ngimel @yf225 @aakhundov @chenyang78

peterbell10 committed:
[inductor] Fix no-xdim reductions

Fixes #103481

Normally triton tensors have shape `[XBLOCK, RBLOCK]`, or some variation where
the lengths are 1 but the number of dimensions is the same. The `no_x_dim`
change in addition to removing the x dimension, also removed the r dimension
from certain values such as the results of reductions and the `xindex` variable.

This fixes those two cases to correctly produce tensors of shape `[1]`,
analogous to the shape `[XBLOCK, 1]` that they would have in the x dim present
case.

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Fix no-xdim reductions"


Fixes #103481

Normally triton tensors have shape `[XBLOCK, RBLOCK]`, or some variation where
the lengths are 1 but the number of dimensions is the same. The `no_x_dim`
change in addition to removing the x dimension, also removed the r dimension
from certain values such as the results of reductions and the `xindex` variable.

This fixes those two cases to correctly produce tensors of shape `[1]`,
analogous to the shape `[XBLOCK, 1]` that they would have in the x dim present
case.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 aakhundov chenyang78

[ghstack-poisoned]

peterbell10 committed:
Update on "[inductor] Fix no-xdim reductions"


Fixes #103481

Normally triton tensors have shape `[XBLOCK, RBLOCK]`, or some variation where
the lengths are 1 but the number of dimensions is the same. The `no_x_dim`
change in addition to removing the x dimension, also removed the r dimension
from certain values such as the results of reductions and the `xindex` variable.

This fixes those two cases to correctly produce tensors of shape `[1]`,
equivalent to the old shape `[XBLOCK, 1]` with the x-dimension dropped.

[ghstack-poisoned]

ngimel reviewed (approved):
Does this still preserve good perf?

peterbell10 commented:
@pytorchbot merge

peterbell10 commented:
@pytorchbot merge

peterbell10 committed:
Update on "[inductor] Fix no-xdim reductions"


Fixes #103481

Normally triton tensors have shape `[XBLOCK, RBLOCK]`, or some variation where
the lengths are 1 but the number of dimensions is the same. The `no_x_dim`
change in addition to removing the x dimension, also removed the r dimension
from certain values such as the results of reductions and the `xindex` variable.

This fixes those two cases to correctly produce tensors of shape `[1]`,
equivalent to the old shape `[XBLOCK, 1]` with the x-dimension dropped.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy ngimel yf225 aakhundov chenyang78

[ghstack-poisoned]

peterbell10 commented:
@pytorchbot merge
